test/pkg/environment/azure/environment.go

```diff
@@ -40,16 +44,57 @@ const (
 
 type Environment struct {
 	*common.Environment
-	Region string
+	AzureClients
+	Vars
+}
+
+type Vars struct {
+	NodeResourceGroup string
+	Region            string
+	SubscriptionID    string
+	VNETResourceGroup string
+	ACRName           string
+	ClusterName       string
+	ClusterResourceGroup string
+}
+type AzureClients struct {
+	VNETClient       *armnetwork.VirtualNetworksClient
+	InterfacesClient *armnetwork.InterfacesClient
 }
 
 func NewEnvironment(t *testing.T) *Environment {
 	env := common.NewEnvironment(t)
-
-	return &Environment{
-		Region:      "westus2",
+	azureEnv := &Environment{
 		Environment: env,
 	}
+	azureEnv.ClusterName = os.Getenv("CLUSTER_NAME")
+	azureEnv.ClusterResourceGroup = os.Getenv("AZURE_RESOURCE_GROUP")
+	azureEnv.ACRName = os.Getenv("ACR_NAME")
+	azureEnv.Region = os.Getenv("AZURE_LOCATION")
```

Comment:
Should we align these values with karpenter options? We could import the type, although there is a subset of options required by e2e, that are not used by our karpenter provider(ACR_NAME). 

For now im proposing we should just leave as is rather than trying to mutate our options type, but let me know if anyone has opinions on this.